### PR TITLE
Hook up user authentication on the frontend

### PR DESCRIPTION
**Pull request by eanile**
Time taken: 2 days

This involves correctly registering users in the database through the frontend and proper navigation to the main page from register and login screens.

Also investigate how to maintain a currently logged in user (Redux allows me to store it, but I don't know if it persists after the app is closed and reopened).

When logging in with incorrect credentials:
image

When signing up for an account with an email that's already associated with an account:
image

### CODE REVIEW

File: PlateUp-flask/run.py [hunk 1]
```diff
@@ -73,8 +73,7 @@ def post(self):
         db.session.add(new_user)
```
Comment (fcpranav) [inline PlateUp-flask/run.py:73]: These changes seem to be same as master? I suggest rebase on master, will make the diff much cleaner.

File: PlateUp-reactnative/screens/Login.js [hunk 2]
```diff
@@ -1,23 +1,58 @@
+import axios from "axios";
+import { Button, Icon, Input } from "../components";
+import { argonTheme, Images } from "../constants";
+import { LinearGradient } from "expo-linear-gradient";
+import { Block, Text } from "galio-framework";
```
Comment (fcpranav) [inline PlateUp-reactnative/screens/Login.js]: Thanks for sorting this alphabetically 👍

File: PlateUp-reactnative/screens/Login.js [hunk 3]
```diff
@@ -1,23 +1,58 @@
...
 
 class Login extends React.Component {
-  render() {
-    const { navigation } = this.props;
+  state = {
+    email: "",
+    password: "",
+  }
 
+  handleLogin = (navigation) => {
```
Comment (fcpranav) [inline PlateUp-reactnative/screens/Login.js]: Doesn't need navigation argument, the function already has access to props.

File: PlateUp-reactnative/screens/Register.js [hunk 4]
```diff
@@ -1,20 +1,71 @@
...
+            navigation.navigate("App");
+          })
+          .catch(() => {
+            util.toast("Login failed! Please confirm that the email and password are correct.");
+          });
+      })
+      .catch(err => {
+        if (err.response.status === 409) {
+          util.toast(`The user with email ${this.state.email} already exists. Please log in instead.`);
+        }
```
Comment (fcpranav) [inline PlateUp-reactnative/screens/Register.js:74]: Should add in a default error such as "Internal Server Error" for cases where the server returns 500. Would make development easier + in the case that there was a different error, the user would still be informed something is wrong.

File: PlateUp-reactnative/util.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { AlertIOS, ToastAndroid } from "react-native";
+
+export const SERVER_URL = "http://192.168.0.18:5000";
```
Comment (fcpranav) [inline PlateUp-reactnative/util.js]: Can we add a note in the documentation on how to configure this? Perhaps we can use an environment variable for this.

**Comment by fcpranav** [comment]
Overall looks good, minor changes requested, some general feedback that does not need to be addressed:
- Switching over to react-redux will probably be beneficial to us when handling recipe data, for now, let's keep this implementation of just redux. In the future, we can port over if we end up switching. When it comes to API calls, it is also useful to introduce more states.  For instance, showing a loading spinner while the API call is processing. The user api should be relatively fast though.
- Outside of the scope of the course, but using a mocking library would be great here. Mocking libraries let us mock out API calls so that we do not need to rely on our backend to be running. Might be worth doing later on if we have extra time.

File: PlateUp-reactnative/screens/Register.js (hunk 4 above)
Comment (fcpranav) [reply PlateUp-reactnative/screens/Register.js:74]: Similarly for Login.js

File: PlateUp-flask/run.py (hunk 1 above)
Comment (eanile) [reply PlateUp-flask/run.py:73]: Yeah not sure what happened when I merged, pretty sure I rebased. Hopefully it doesn't happen next time.

File: PlateUp-reactnative/screens/Register.js (hunk 4 above)
Comment (eanile) [reply PlateUp-reactnative/screens/Register.js:74]: Done. Also added some default "unknown error occurred" if for some reason its not even a 500 that occurs.

File: PlateUp-reactnative/screens/Login.js (hunk 3 above)
Comment (eanile) [reply PlateUp-reactnative/screens/Login.js]: Done.

File: PlateUp-reactnative/util.js (hunk 5 above)
Comment (eanile) [reply PlateUp-reactnative/util.js]: I changed it to be an env.js file that we import and use. That file is excluded from the project and I will add to the README describing the format of the file. I tried using libraries that take advantage of .env files but they were all causing obscure errors that I couldn't find fixes for so I settled for a js file.

**Comment by eanile** [comment]
According to the redux website, the createStore function, which is the only function I import from redux, belongs to redux. react-redux is used for other stuff. So I'm going to leave it as redux for now.

File: .gitignore [hunk 6]
```diff
@@ -1,15 +1,23 @@
-# Expo
-PlateUp-reactnative/.expo
-PlateUp-reactnative/.expo-shared
+### React Native ###
+PlateUp-reactnative/node_modules/**/*
+PlateUp-reactnative/.expo/*
+PlateUp-reactnative/.expo-shared/*
+PlateUp-reactnative/npm-debug.*
+PlateUp-reactnative/package-lock.json
```
Comment (fcpranav) [inline .gitignore]: This shouldn't be in gitignore, the package-lock.json should be committed. It is basically what ensures everyone has the same versions running.

File: PlateUp-reactnative/util.js (hunk 5 above)
Comment (fcpranav) [reply PlateUp-reactnative/util.js]: Okay sure, we can revisit this later if needed. Not a biggie.

File: .gitignore [hunk 7]
```diff
@@ -1,15 +1,24 @@
...
+PlateUp-reactnative/yarn.lock
```
Comment (fcpranav) [inline .gitignore:6]: yarn.lock also probably doesn't belong here.

Comment (eanile) [reply .gitignore:6]: Will leave it in for now.

File: .gitignore (hunk 6 above)
Comment (eanile) [reply .gitignore]: Done.

**Review by fcpranav — APPROVED** [review]
LGTM

**Pull request merged**